client/my-sites/themes/install-theme-button.tsx

```diff
@@ -1,18 +1,27 @@
 import { FEATURE_INSTALL_THEMES } from '@automattic/calypso-products';
 import { Button } from '@automattic/components';
 import { translate } from 'i18n-calypso';
-import { connect } from 'react-redux';
+import { connect, ConnectedProps } from 'react-redux';
+import { Dispatch } from 'redux';
 import { recordTracksEvent } from 'calypso/state/analytics/actions';
 import { isUserLoggedIn } from 'calypso/state/current-user/selectors';
 import isAtomicSite from 'calypso/state/selectors/is-site-automated-transfer';
 import siteHasFeature from 'calypso/state/selectors/site-has-feature';
 import { isJetpackSiteMultiSite, isJetpackSite } from 'calypso/state/sites/selectors';
+import { IAppState } from 'calypso/state/types';
 import { getSelectedSiteId, getSelectedSiteSlug } from 'calypso/state/ui/selectors';
 import { trackClick } from './helpers';
-
 import './install-theme-button.scss';
 
-const InstallThemeButton = ( {
+interface TracksEventProps {
+	tracksEventProps: { site_type: string | null };
+}
+
+interface InstallThemeButtonProps extends PropsFromRedux {
+	canUploadThemesOrPlugins: boolean;
+}
+
+const InstallThemeButton: React.FC< InstallThemeButtonProps > = ( {
```

Comment:
in general it's good to avoid `React.FC` and some of the other React component types. They are riddled with problems related to `children` and `ref`s and such. It seems counter-intuitive, but the types were never right to begin with. In some projects people create their own base component type, but here it's probably easiest and most pragmatic simply to type this as a function expecting those props and returning something like `JSX.Element`

Reply:
Since React 18 and the corresponding types, `children` is no longer an implicit prop. So, I think it's completely fine to use `React.FC`.

Reply:
https://github.com/Automattic/wp-calypso/blob/trunk/package.json#L195

at some point we'll be on 18, but I don't know if we should make decisions about code today based on where we hope to be tomorrow